### PR TITLE
fix: fix wrong import on annotated parameter #3320

### LINKED ISSUE #3320: Wrong javax import statement causing compilation error
Hello,

thank you for this great tool.

The generated code for a class that declares a javax validation annotation on its parameter has wrong import statement which causes compile time errors. This only seems to be an issue with javax annotations such as @Valid and @NotNull

for a declaration of 
```java
public void testThatValidAnnotationImportsAreCreated(@Valid @NotNull RandomDto randomDto) 
{
        
throw new IllegalArgumentException("An Error has occurred");

 }
```

The import generated is

```java
import com.example.processor.src.other.@javax.validation.Valid

@javax.validation.constraints.NotNull

RandomDto;
```

instead of 

```java
import com.example.processor.src.other.RandomDto;

import javax.validation.Valid;

import javax.validation.constraints.NotNull;
```

this is related to issue #1773 and #1774

Tested on Spoon 8.1.0 and 8.2.0-beta2 running java 8 

My launcher config is

``` java
  SpoonAPI launcher = new Launcher();

  Environment environment = launcher.getEnvironment();

  environment.setNoClasspath(true);

  environment.setAutoImports(true);

  launcher.addInputResource(s);

  launcher.setSourceOutputDirectory(sourceOutputDirectory);

  ClassProcessor processor = new ClassProcessor();

  launcher.addProcessor(processor);

  launcher.process();

  launcher.prettyprint();
  ```

### PR DESCRIPTION
**Pull request by gibahjoe**
fix #3320

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -159,6 +159,14 @@
       <artifactId>maven-invoker</artifactId>
       <version>3.0.1</version>
     </dependency>
+    <!-- https://mvnrepository.com/artifact/javax.validation/validation-api -->
```
Comment (monperrus) [inline pom.xml:162]: The test is in noclasspath mode. Remove this or work in fullclasspath?

File: src/test/java/spoon/test/imports/ImportTest.java [hunk 2]
```diff
@@ -1777,4 +1778,22 @@ public void testImportByJavaDoc() throws Exception {
 		}
 	}
 
+	@Test
+	public void testThatCorrectImportsAreGeneratedForJavaxAnnotatedElements() {
```
Comment (monperrus) [inline src/test/java/spoon/test/imports/ImportTest.java:1782]: In Spoon, the tests come with a one-line contract describing the test intention, something like

// contract: annotations are not added on import statements

could you add the contract that is tested? Thanks!

File: src/test/java/spoon/test/imports/ImportTest.java [hunk 3]
```diff
@@ -1777,4 +1778,22 @@ public void testImportByJavaDoc() throws Exception {
...
+		e.setNoClasspath(true);
+		e.setAutoImports(true);
+		l.addInputResource("src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/TestSource.java");
+		l.run();
+
+		CtType<TestSource> objectCtType = l.getFactory().Type().get(TestSource.class);
+		CompilationUnit compilationUnit = l.getFactory().CompilationUnit().getOrCreate(objectCtType);
+
+		for (CtImport anImport : compilationUnit.getImports()) {
+			System.out.println(anImport.prettyprint());
```
Comment (monperrus) [inline src/test/java/spoon/test/imports/ImportTest.java:1795]: Remove System.out from tests.

**Comment by gibahjoe** [comment]
Thanks.I have attended to  the pr comments

File: src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/TestSource.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package spoon.test.imports.testclasses.badimportissue3320.source;
+
+import spoon.test.imports.testclasses.badimportissue3320.source.other.SomeObjectDto;
+import spoon.test.imports.testclasses.badimportissue3320.source.other.TestAnnotation;
+
+/**
+ * @author Gibah Joseph
+ * Email: gibahjoe@gmail.com
+ * Apr, 2020
+ **/
```
Comment (MartinWitt) [inline src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/TestSource.java:10]: Personally i hate reading this. It feels like useless information, which get outdated easily. Git can provide this information better. Maybe remove it?

File: src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/other/SomeObjectDto.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package spoon.test.imports.testclasses.badimportissue3320.source.other;
+
+/**
+ * @author Gibah Joseph
+ * Email: gibahjoe@gmail.com
+ * Apr, 2020
+ **/
```
Comment (MartinWitt) [inline src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/other/SomeObjectDto.java:7]: see other comment

File: src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/other/TestAnnotation.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * @author Gibah Joseph
+ * email: gibahjoe@gmail.com
+ * Apr, 2020
+ **/
```
Comment (MartinWitt) [inline src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/other/TestAnnotation.java:13]: see other comment

File: src/test/java/spoon/test/imports/ImportTest.java [hunk 7]
```diff
@@ -1777,4 +1778,20 @@ public void testImportByJavaDoc() throws Exception {
...
+	public void testImportsForElementsAnnotatedWithTypeUseAnnotations() {
+		// contract: correct import generated for method parameters annotated with TYPE_USE annotations
+		final Launcher l = new Launcher();
+		Environment e = l.getEnvironment();
+
+		e.setNoClasspath(true);
+		e.setAutoImports(true);
+		l.addInputResource("src/test/java/spoon/test/imports/testclasses/badimportissue3320/source/TestSource.java");
+		l.run();
+
```
Comment (MartinWitt) [inline src/test/java/spoon/test/imports/ImportTest.java:1791]: maybe a pit "picky", but short names like e and l makes reading hard.

File: src/test/java/spoon/test/imports/ImportTest.java [hunk 8]
```diff
@@ -1777,4 +1778,20 @@ public void testImportByJavaDoc() throws Exception {
...
+		CtType<TestSource> objectCtType = l.getFactory().Type().get(TestSource.class);
+		CompilationUnit compilationUnit = l.getFactory().CompilationUnit().getOrCreate(objectCtType);
+
+		assertEquals(1, compilationUnit.getImports().stream().filter(ctImport -> ctImport.prettyprint().equals("import spoon.test.imports.testclasses.badimportissue3320.source.other.SomeObjectDto;")).count());
```
Comment (MartinWitt) [inline src/test/java/spoon/test/imports/ImportTest.java:1795]: maybe a linebreak, but not mandatory

**Comment by MartinWitt** [comment]
LGTM, great work

**Comment by monperrus** [comment]
Thanks a lot and congrats on your first contribution!

**Comment by gibahjoe** [comment]
Thanks

**Pull request merged**